src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs

```diff
@@ -638,13 +648,24 @@ internal static bool LocalFindAndPop(object callback)
 
         public object? Dequeue(ThreadPoolWorkQueueThreadLocals tl, ref bool missedSteal)
         {
+            ThreadPoolWorkQueue workQueue = ThreadPool.s_workQueue;
```

Comment:
This shouldn't be necessary, the method is already an instance method of ThreadPoolWorkQueue